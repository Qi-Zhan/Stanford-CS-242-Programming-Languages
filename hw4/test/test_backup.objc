START VERIFY;

o = [ retrieve = \x.x, backup = \x. x.retrieve <- \y.x ];
o.retrieve;
o1 = o.backup;
o2 = o1.newdata <- \o.o;
o2.retrieve;

STOP VERIFY;
