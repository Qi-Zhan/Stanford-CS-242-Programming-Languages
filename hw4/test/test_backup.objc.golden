o = [backup = \x.((x.retrieve <- \y.(x))), retrieve = \x.(x)];
[backup = \x.((x.retrieve <- \y.(x))), retrieve = \x.(x)];
o1 = [backup = \x.((x.retrieve <- \y.(x))), retrieve = \y.([backup = \x.((x.retrieve <- \y.(x))), retrieve = \x.(x)])];
o2 = [backup = \x.((x.retrieve <- \y.(x))), newdata = \o_0.(o_0), retrieve = \y.([backup = \x.((x.retrieve <- \y.(x))), retrieve = \x.(x)])];
[backup = \x.((x.retrieve <- \y_0.(x))), retrieve = \x.(x)];
